pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -213,6 +230,67 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
 	return identity, false, nil
 }
 
+// Sets the finalizers for UserAccount
+func (r *ReconcileUserAccount) setFinalizers(userAcc *toolchainv1alpha1.UserAccount) error {
+	// Add the finalizer if it is not present
+	if !containsString(userAcc.ObjectMeta.Finalizers, userAccFinalizerName) {
+		userAcc.ObjectMeta.Finalizers = append(userAcc.ObjectMeta.Finalizers, userAccFinalizerName)
+		if err := r.client.Update(context.TODO(), userAcc); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Handles finalizer logic for UserAccount. Returns bool to indicate Requeue
+func (r *ReconcileUserAccount) finalizer(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (bool, error) {
+	logger.Info("deleting UserAccount and subsequent resources", "name", userAcc.Name)
+
+	// Get the Identity associated with the UserAccount
+	identity := &userv1.Identity{}
+	identityName := ToIdentityName(userAcc.Spec.UserID)
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: identityName}, identity); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return false, nil
+		}
+		return false, err
+	}
+
+	// Get the User associated with the UserAccount
+	user := &userv1.User{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return false, nil
+		}
+		return false, err
+	}
+
+	// Delete Identity associated with UserAccount
+	if err := r.client.Delete(context.TODO(), identity); err != nil {
+		return true, err
```

Comment:
if error is not_nil, no need to set requeue to true.  check the doc of `Reconcile()` method

```
// The Controller will requeue the Request to be processed again if the returned error is non-nil or
// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
```

Reply:
@nurali-techie Fixed here: 51e17d72b87aa88373f9bdb8dd5dd696b2b29847